### ISSUE TITLE
npm false positive CVEs

### ISSUE THREAD

**Issue by thediveo** [anchore/grype#496]
More false positives turned up in the web UI part of my [lxkns](https://github.com/thediveo/lxkns) project.

```yaml
ignore:
    # Barking up the wrong tree...
    - vulnerability: CVE-2009-4590 # php, not npm package
      package:
        type: npm
    - vulnerability: CVE-2009-4591 # sql engine, not npm package
      package:
        type: npm
    - vulnerability: CVE-2009-4592 # php, not npm package
      package:
        type: npm
    - vulnerability: CVE-2011-0220 # apfel bonjour, not npm package
      package:
        type: npm
    - vulnerability: CVE-2013-1779 # drupal, not npm package
      package:
        type: npm
    - vulnerability: CVE-2014-2980 # gnustep, not npm package
      package:
        type: npm
    - vulnerability: CVE-2017-18589 # rust, not npm package
      package:
        type: npm
    - vulnerability: CVE-2020-7663 # ruby, not npm package
      package:
        type: npm
    - vulnerability: CVE-2020-7753 # component/trim, not Trott/trim
      package:
        type: npm
```

**Comment by luhring** [comment]
Thanks @TheDiveO! Keep 'em coming.

**Comment by TheDiveO** [comment]
And another one, where the CVE origins can't get their act correctly and make another mess: a match of `CVE-2015-5237` on google.golang.org/protobuf ... which actually is the golang version, where the CVE seems to refer to https://github.com/protocolbuffers/protobuf, see also the CVE reference to https://github.com/protocolbuffers/protobuf/issues/760 which is the C version, but not the Go implementation.

**Comment by westonsteimel** [comment]
> And another one, where the CVE origins can't get their act correctly and make another mess: a match of `CVE-2015-5237` on google.golang.org/protobuf ... which actually is the golang version, where the CVE seems to refer to https://github.com/protocolbuffers/protobuf, see also the CVE reference to [protocolbuffers/protobuf#760](https://github.com/protocolbuffers/protobuf/issues/760) which is the C version, but not the Go implementation.

@luhring, one thing that might help for Go in particular would be to make use of the information coming from [The Go Vulnerability Database](https://github.com/golang/vulndb) and the corresponding go module [x/vuln](https://pkg.go.dev/golang.org/x/vuln).  In this case CVE-2015-5237 is already identified as a false positive there: https://github.com/golang/vuln/blob/f553aa4445846cef9869d51edc1da637ea30bb77/triaged-cve-list#L29

**Comment by TheDiveO** [comment]
Yet the false positives I reported here are mostly hitting the npm world, except for the lately mentioned protobuf-related Go mod false positive. I find it slightly worrysome that all ecosystems will/would need to maintain their own false positive "defense shields" to deflect the sometimes mediocre quality of NVD/* originating CVE information, where there seems to be a fondness of using "*" much too often and thus basically crying "wolf" all the time on world+dog.

**Comment by luhring** [comment]
> > And another one, where the CVE origins can't get their act correctly and make another mess: a match of `CVE-2015-5237` on google.golang.org/protobuf ... which actually is the golang version, where the CVE seems to refer to https://github.com/protocolbuffers/protobuf, see also the CVE reference to [protocolbuffers/protobuf#760](https://github.com/protocolbuffers/protobuf/issues/760) which is the C version, but not the Go implementation.
> 
> @luhring, one thing that might help for Go in particular would be to make use of the information coming from [The Go Vulnerability Database](https://github.com/golang/vulndb) and the corresponding go module [x/vuln](https://pkg.go.dev/golang.org/x/vuln). In this case CVE-2015-5237 is already identified as a false positive there: https://github.com/golang/vuln/blob/f553aa4445846cef9869d51edc1da637ea30bb77/triaged-cve-list#L29

This is a great idea @westonsteimel, thanks! 👍 

> I find it slightly worrysome that all ecosystems will/would need to maintain their own false positive "defense shields"

I don't think I understand this fully, but I don't think it's necessary for all ecosystems to maintain the same kind of list. Here we have a problem, where the largest bottleneck is dirty data. Ideally, all ecosystem's vulnerability data is in good shape, but in practice it's not. Given that, I'd say consumers of this data (like Grype) should take any help they can get with opportunities to draw better conclusions from the data.

**Comment by TheDiveO** [comment]
What I'm referring here to is "cross polution", or in older terms, "clan liability", due to what you term "dirty data". As we won't have an ideal world, this means that a single act truely related to only one package in one ecosystem now causes lots of additional work in not only another, but even in several cases all other ecosystems. And that's because the database origin does sloppy work, causing lot of unnecessary work, while potentially easing its own work due to less work on initial due dilligence. This is unbelieveable more so as the running text of CVEs often correctly state the ecosystem or at least clearly hint at it. This sloppy upstream work then causes lots of friction and finally refusal with the real work horses, the developers; all this significantly reducing acceptance of security tools. And then we end up with security lip service.

**Comment by spiffcs** [comment]
@thediveo thanks again for filing this issue. We've done a ton of work recently regarding corrections in the data source as well as db corrections for npm and golang modules. Would you be able to provide a source we can scan that reproduces these FP if they're still valid?

Thanks again!

**Comment by thediveo** [comment]
@spiffcs https://github.com/thediveo/lxkns should suffice as a test, as it is the guinea pig repo where I tripped over the messed-up CVEs when letting syft+grype on the loose.

**Comment by OfriOuzan** [comment]
We encountered the same issue on the following environment
**What happened**:
In a Vulnerability Scanner Benchmark Research we are conducting, we executed Grype on 20 different containers and found out that Grype has multiple False Positives.
**What you expected to happen**:
We expected Grype not to report on these CVEs.
**How to reproduce it (as minimally and precisely as possible)**:
Install the Docker Images (from the links below) and execute Grype using the following command:
grype <container_name> —-output json > <output_file_path>

- **Output of `grype version`**:
Application:          grype
Version:              0.41.0
Syft Version:         v0.50.0
BuildDate:            2022-07-06T15:20:18Z
GitCommit:            0e0a9d9e7a28592db489499db0294608e5fe69b8
GitDescription:       v0.41.0
Platform:             linux/amd64
GoVersion:            go1.18.3
Compiler:             gc
Supported DB Schema:  4
**Ghost**
 
- **Container Details**:
https://hub.docker.com/layers/library/ghost/5.2.4/images/sha256-42137b9bd1faf4cdea5933279c48a912d010ef614551aeb0e44308600aa3e69f?context=explore

- **OS (e.g: `cat /etc/os-release`)**:
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
- **CVEs**
**CVE-2009-4590, CVE-2009-4591, CVE-2009-4592**
Grype wrongly identified CVE-2009-4590, CVE-2009-4591, CVE-2009-4592 as vulnerable.
The path it identified is:
/var/lib/ghost/versions/5.2.4/node_modules/base/package.json
The base package version is 0.11.2
However, according to the debian website, the vulnerability is related to[ acidbase package](https://github.com/jbouse-debian/acidbase/blob/master/base_local_rules.php) which is not exists on the container.

**Comment by willmurphyscode** [comment]
Added `changelog-ignore` because this was fixed in `0.60.0` and so shouldn't be included in the current release's release notes.
